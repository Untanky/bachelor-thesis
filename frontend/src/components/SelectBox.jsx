import React from 'react';
import PropTypes, { string } from 'prop-types';

import './SelectBox.scss';

const SelectBox = ({ selectItems, onChange }) => {
  return (
    <select 
      className="select-box"
      onChange={(evt) => onChange(evt.target.value)}
    >
      { selectItems.map((selectItem) => 
          <option 
            key={Math.random()}
            value={selectItem.value}
          >
            {selectItem.text}
          </option>
      ) }
    </select>
  )
};

SelectBox.propTypes = {
  selectItems: PropTypes.arrayOf(
    PropTypes.shape({
      value: PropTypes.string,
      test: PropTypes.string,
    })
  ).isRequired,
  onChange: PropTypes.func.isRequired,
}

export default SelectBox;
