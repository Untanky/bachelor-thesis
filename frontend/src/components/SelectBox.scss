@import '../theme';

.select-box {
  background-color: $htw-green;
  color: white;
  font-size: 16px;
  outline: none;
  padding: 0.5em 1em;
  border-radius: 2em;
  border: none;
  box-shadow: 0 4px 4px 0 rgba(0, 0, 0, 0.2);
  transition: all 0.25s ease-in-out;

  &:hover {
    cursor: pointer;
    box-shadow: 0 2px 4px 0 rgba(0, 0, 0, 0.15);
    transition: all 0.25s ease-in-out;
  }

  option {

  }
}