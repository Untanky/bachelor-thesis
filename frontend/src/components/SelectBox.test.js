import React from 'react';
import SelectBox from './SelectBox';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer
    .create(<SelectBox />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
