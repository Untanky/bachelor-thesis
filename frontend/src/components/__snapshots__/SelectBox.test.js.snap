// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<select
  className="select-box"
  onChange={[Function]}
>
  <option
    value="1"
  >
    One
  </option>
  <option
    value="2"
  >
    Two
  </option>
  <option
    value="3"
  >
    Three
  </option>
</select>
`;
